File: Frontend/src/models/order.model.ts
import { Product } from "./product.model";

export type OrderRequest = {
  address: string;
  phone: string;
  paymentMethod: string;
  couponCodes?: string[];
  rewardPointsUsed?: number;
};

export type OrderStatus =
  | "NEW"
  | "CONFIRMED"
  | "PREPARING"
  | "SHIPPING"
  | "DELIVERED"
  | "CANCELLED"
  | "REQUEST_CANCEL";

export type OrderItem = {
  id: number;
  quantity: number;
  price: number;
  product: Product;
  review?: {
    rating: number;
    comment: string;
  } | null;
};

export type Order = {
  id: number;
  orderDate: string;
  totalPrice: number;
  status: OrderStatus;
  paymentMethod: string;
  shippingAddress: string;
  shippingPhone: string;
  orderItems: OrderItem[];
};

export type CashFlowStat = {
  status: string;
  totalAmount: number;
  orderCount: number;
};

File: manager/lib/features/orders/models/order_model.dart
class OrderModel {
  final int id;
  final double totalPrice;
  final String status;
  final String? paymentMethod;
  final String? orderDate;
  final String? shippingAddress;
  final String? shippingPhone;

  OrderModel({
    required this.id,
    required this.totalPrice,
    required this.status,
    this.paymentMethod,
    this.orderDate,
    this.shippingAddress,
    this.shippingPhone,
  });

  factory OrderModel.fromJson(Map<String, dynamic> json) {
    return OrderModel(
      id: json['id'] ?? 0,
      totalPrice: (json['totalPrice'] as num?)?.toDouble() ?? 0.0,
      status: json['status'] ?? 'UNKNOWN',
      paymentMethod: json['paymentMethod'],
      orderDate: json['orderDate'],
      shippingAddress: json['shippingAddress'],
      shippingPhone: json['shippingPhone'],
    );
  }
}
